third_party/terraform/utils/import.go

```diff
@@ -25,7 +26,37 @@ func parseImportId(idRegexes []string, d TerraformResourceData, config *Config)
 			// Starting at index 1, the first match is the full string.
 			for i := 1; i < len(fieldValues); i++ {
 				fieldName := re.SubexpNames()[i]
-				d.Set(fieldName, fieldValues[i])
+				// This part looks confusing.  Because there is no way to know at
```

Comment:
Shrunk comment.  Still need to explain why it's not a try/catch pattern because that will help future code archaeologists avoid making mistakes while refactoring.